astropy/cosmology/core.py

```diff
@@ -4,21 +4,25 @@
 
 import abc
 import inspect
-from typing import TYPE_CHECKING, Any, TypeVar
+from itertools import chain
+from types import MappingProxyType
+from typing import TYPE_CHECKING, Any, ClassVar, TypeVar
 
 import numpy as np
 
 from astropy.io.registry import UnifiedReadWriteMethod
 from astropy.utils.decorators import classproperty
 from astropy.utils.metadata import MetaData
 
+from ._utils import all_cls_vars
 from .connect import (
     CosmologyFromFormat,
     CosmologyRead,
     CosmologyToFormat,
     CosmologyWrite,
 )
 from .parameter import Parameter
+from .parameter._descriptors import DerivedParameterDescriptor, ParameterDescriptor
```

Comment:
Should these not be public, at least within `parameter`? Importing from private modules outside of the module they are in seems weird. (Not relevant for this PR otherwise)

Reply:
I was hesitant to make it public. It fits most naturally within `parameter` but is required here.

Reply:
Yeah, makes sense. Let's leave it as you have it for now.